guides/quickstart.md

```diff
@@ -4,426 +4,459 @@ title: Quickstart
 
 ```js
-import { moduleForComponent, test } from 'ember-qunit';
-import hbs from 'htmlbars-inline-precompile';
+import { hasClass } from 'ember-cli-page-object';
 
-moduleForComponent('login-form', 'Integration | login form', {
-  integration: true
-});
+export default {
+  scope: 'form',
 
-test('calls submit action with correct username and password', function(assert) {
-  assert.expect(2);
+  username: {
+    scope: '[data-test-username]'
 
-  function submit(username, password) {
-    assert.equal(username, 'admin');
-    assert.equal(password, 'secret');
-  }
+    // Add `hasError` boolean property
+    hasError: hasClass('has-error'),
 
-  this.set('submit', submit);
+    errorMessage: {
+      scope: '.error-message'
+    }
+  },
 
-  this.render(hbs`
-    {{login-form
-      submit=(action submit)
-    }}
-  `);
+  // ...
+}
+```
 
-  $username = this.$('#username');
-  $password = this.$('#password');
+Now `hasError` can be used as a `username` getter:
 
-  $username.val('admin');
-  $username.trigger('input');
-  $username.change();
+```js
+assert.ok(loginForm.username.hasError);
+```
 
-  $password.val('secret');
-  $password.trigger('input');
-  $password.change();
+The last missing part is filling an input with a value.
 
-  this.$('button').click();
-});
+As mentioned above there is a built-in `fillIn` component property supplied for each component. Howerver if we call `fillIn` on the `username` right now it would fail because the `username` comforms to the `div[data-test-username]` which can't be filled in. We need instruct `fillIn` to deal with an `input` children node.
 
-test('shows errors', function(assert) {
-  assert.expect(2);
+```js
+import { hasError, fillable } from 'ember-cli-page-object';
 
-  this.set('errors', []);
+export default {
+  scope: 'form',
 
-  this.render(hbs`
-    {{login-form
-      errors=errors
-    }}
-  `);
+  username: {
+    scope: '[data-test-username]'
 
-  assert.equal(this.$('.errors').trim().text()), '');
+    hasError: hasClass('has-error'),
 
-  Ember.run(() => {
-    this.set('errors', ['Invalid credentials']);
-  });
+    // All the parent scopes are getting prepended to the final selector
+    // so the final selector would become "form [data-test-username] input"
+    fillIn: fillable('input'),
 
-  assert.equal(this.$('.errors').trim().text()), 'Invalid credentials');
-});
+    errorMessage: {
+      scope: '.error-message'
+    }
+  },
+
+  // ...
+}
 ```
 
-Let's use our existing page object to refactor these integration tests. As a reminder, here is our page object. (We don't need to change anything to use it in our integration tests!)
+That's it. Our `username` is ready to be used in the tests. But what about the `password`? Should we copy-paste all the `username` implementation into it?
+
+## Macroses
+
+Well, the only difference between  the `username` and `password` is a `scope` selector. It could be annoying to repeat the whole field definition accross all the similar fields. 
```

Comment:
Seems like `login-form` is too complex for the Quickstart page. I think user defined "macroses" better fits to the "Extensibility" page.